src/app/space/settings/areas/areas.component.ts

```diff
@@ -18,69 +30,251 @@ export class AreasComponent implements OnInit, OnDestroy {
   @ViewChild(CreateAreaDialogComponent) createAreaDialog: CreateAreaDialogComponent;
   @ViewChild(ModalDirective) modal: ModalDirective;
 
+  private allAreas: Area[]; // flat array obtained directly from API
+  private filteredAreas: Area[]; // flat array filtered and sorted
+  private treeAreas: Area[]; // transformed flat array into tree list
```

Comment:
Looks like `treeAreas` is updated but it doesn't get read. Did you forget to use it or just not need it?

Reply:
It's used in the HTML template by the pfng-tree-list component